testkit/src/api.rs

```diff
@@ -147,338 +98,191 @@ impl TestKitApi {
-    mut response: Response,
-) -> IronResult<Response> {
-    use iron::headers::ContentLength;
-    use iron::method::Method;
-    use iron::response::WriteBody;
-
-    fn has_body(method: &Method) -> bool {
-        match *method {
-            Method::Get | Method::Head | Method::Delete | Method::Trace => false,
-            _ => true,
+        match response.status() {
+            StatusCode::Ok => Ok({
+                let body = response.text().expect("Unable to get response text");
+                trace!("Body: {}", body);
+                serde_json::from_str(&body).expect("Unable to deserialize body")
+            }),
+            StatusCode::Forbidden => Err(api::Error::Unauthorized),
+            StatusCode::BadRequest => Err(api::Error::BadRequest(error(response))),
+            StatusCode::NotFound => Err(api::Error::NotFound(error(response))),
+            s if s.is_server_error() => Err(api::Error::InternalError(error(response).into())),
+            s => panic!("Received non-error response status: {}", s.as_u16()),
         }
     }
-
-    if !log_enabled!(Level::Trace) {
-        // Avoid expensive string allocations.
-        return Ok(response);
-    }
-
-    let mut url = request
-        .extensions
-        .get::<OriginalUrl>()
-        .unwrap_or(&request.url)
-        .path()
-        .join("/");
-    if let Some(query) = request.url.query() {
-        url += "?";
-        url += query;
-    }
-
-    let req_body: Option<String> = if has_body(&request.method) {
-        request
-            .get::<bodyparser::Raw>()
-            .expect("cannot read request body")
-    } else {
-        None
-    };
-
-    let response_body: Option<String> = {
-        let len: u64 = response
-            .headers
-            .get::<ContentLength>()
-            .map(|len| len.0)
-            .unwrap_or_default();
-
-        response.body.take().map(|mut body| {
-            let mut buffer = Vec::with_capacity(len as usize);
-            body.write_body(&mut buffer)
-                .expect("cannot write response body to buffer");
-            String::from_utf8(buffer).expect("cannot decode response body")
-        })
-    };
-
-    trace!(
-        "{method} ({access}) /{url}{req_body_tag}{req_body}\n\
-         Response: {resp_status}\n{resp_body}\n",
-        method = request.method,
-        access = access,
-        url = url,
-        req_body_tag = if req_body.is_some() { "\nBody: " } else { "" },
-        req_body = req_body.unwrap_or_default(),
-        resp_status = response
-            .status
-            .as_ref()
-            .map(ToString::to_string)
-            .unwrap_or_else(|| "(no status)".to_string()),
-        resp_body = response_body
-            .as_ref()
-            .map(String::as_ref)
-            .unwrap_or("(no body)")
-    );
-
-    // Return the body to the response
-    response.body = response_body.map(|body| Box::new(body) as Box<WriteBody>);
-    Ok(response)
 }
 
-#[cfg(test)]
-mod tests {
```

Comment:
Why all the tests were removed?

Reply:
Because I can't create `reqwest::Response` instance by hands thus I can't port these tests. Instead of this I test `response_to_api_result` behaviour in regular testkit tests.